backend/backend-schema-proposal.md

```diff
@@ -39,59 +39,128 @@ GET `/request/{van_id}`
 GET `/location`
 
 - Subscribes the client to SSE triggered by new coordinates to any van
-- Responses are in the format `(lat, lon, route_id, van_id, timestamp)` where
-  `van_id` & `route_id` are 8-bit ints, `lat` & `lon` are 64-bit doubles, and
-  `timestamp` is a UNIX timestamp
 - Any response with an earlier timestamp than the most recent should be discarded
 - Any timestamp more than 1 min in the future should be discarded
 - Communication can be HTTP or HTTPS, but HTTPS is already being used
+- Response body is defined in JSON as per the following typescript spec:
+
+``ts
+interface Location {
+    vanId: number,
+    routeId: number,
+    location: Coordinate,
+    timestampMillis: number
+}
+``
 
 GET `/routes`
 
 - Retrieves a predefined list of routes and their waypoints
 - Coordinates are defined in a config file
-- Responses are a JSON array of routes in the following format:
+- Response body is defined in JSON as per the following typescript spec:
+
+```typescript
+interface Routes {
+    routes: Route[],
+    stops: Stop[]
+}
+
+interface Route {
+    id: number,
+    name: string,
+    active: boolean,
+    waypoints: Coordinate[] // Already ordered by index in backend
+}
+
+interface Stop {
+    id: number,
+    name: string,
+    active: boolean,
+    location: Coordinate
+}
+```
+
+GET `/alert`
```

Comment:
specify `/alert/stop/{stop_id}` and `/alert/route/{route_id}`

Reply:
Why exactly?